### PR TITLE
Newsletters groups are optional

### PR DESCRIPTION
**Pull request by Fatalito**
## What does this change?
Newsletter groups are optional.
This will allow removing the deprecated category "work", this change will be driven by the newsletter api.
This is the first step to making the newsletter page more dynamic.

## Screenshots
No visual changes.

### Tested

- [X] Locally

### CODE REVIEW

**Comment by Fatalito** [comment]
[Make the grouped newsletter category fields optional on frontend](https://trello.com/c/JxIF4TG5/523-make-the-grouped-newsletter-category-fields-optional-on-frontend)

File: common/app/services/newsletters/NewsletterApi.scala [hunk 1]
```diff
@@ -133,4 +111,26 @@ case class NewsletterApi(wsClient: WSClient)(implicit executionContext: Executio
     }
   }
 
+  private def getBody(path: String): Future[JsValue] = {
```
Comment (DavidLawes) [inline common/app/services/newsletters/NewsletterApi.scala:114]: I think this change just moves code around a bit - let me know if I've missed something!

File: common/app/services/newsletters/NewsletterApi.scala [hunk 2]
```diff
@@ -44,73 +44,51 @@ object GroupedNewsletterResponse {
...
+      newsRoundups,
+      newsByTopic,
+      features,
+      sport,
+      culture,
+      lifestyle,
+      comment,
+      work,
+      fromThePapers,
+    ).flatten.map(g => (g.displayName, g.newsletters))
```
Comment (DavidLawes) [inline common/app/services/newsletters/NewsletterApi.scala:68]: This is a neat change :)

File: common/app/services/newsletters/NewsletterApi.scala [hunk 3]
```diff
@@ -44,73 +44,51 @@ object GroupedNewsletterResponse {
 
 // TODO: Find a better way to define this that means Frontend doesn't have knowledge of the fields returned.
```
Comment (DavidLawes) [inline common/app/services/newsletters/NewsletterApi.scala:45]: Would be interested to hear if you have any thoughts about how this 'TODO' could be addressed (only out of pure curiosity!)

**Review by DavidLawes — APPROVED** [review]
👍  - this change makes sense to me, I can see why an optional group of newsletters will increase flexibility.

File: common/app/services/newsletters/NewsletterApi.scala (hunk 1 above)
Comment (Fatalito) [reply common/app/services/newsletters/NewsletterApi.scala:114]: Yes, second commit is scalafmt, it reordered stuff. It was breaking in CI before.

File: common/app/services/newsletters/NewsletterApi.scala (hunk 2 above)
Comment (Fatalito) [reply common/app/services/newsletters/NewsletterApi.scala:68]: 👍

File: common/app/services/newsletters/NewsletterApi.scala (hunk 3 above)
Comment (Fatalito) [reply common/app/services/newsletters/NewsletterApi.scala:45]: We are thinking of replacing the call to grouped newsletters to a call to newsletters and build the groups on the frontend. Similarly to what is done for /user-prefs

**Comment by Fatalito** [comment]
I don't have merging access, could you do it?

**Pull request merged**